controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentJobs.java

```diff
@@ -55,13 +55,13 @@ private DeploymentJobs(OptionalLong projectId, Map<JobType, JobStatus> status, O
     }
 
     /** Return a new instance with the given completion */
-    public DeploymentJobs withCompletion(long projectId, JobType jobType, JobStatus.JobRun completion, Optional<JobError> jobError) {
+    public DeploymentJobs withCompletion(JobType jobType, JobStatus.JobRun completion, Optional<JobError> jobError) {
         Map<JobType, JobStatus> status = new LinkedHashMap<>(this.status);
         status.compute(jobType, (type, job) -> {
             if (job == null) job = JobStatus.initial(jobType);
             return job.withCompletion(completion, jobError);
         });
-        return new DeploymentJobs(jobType == JobType.component ? OptionalLong.of(projectId) : this.projectId, status, issueId, builtInternally);
+        return new DeploymentJobs(projectId, status, issueId, builtInternally);
```

Comment:
Only the job status members are used here. They're kept for now only for serialisation of old Instance objects.